### PR TITLE
[Merged by Bors] - chore: cleanups following #8609 and #8714

### PR DESCRIPTION
**Pull request by alreadydone**
+ generalize `image_rootSet`, `adjoin_rootSet_eq_range` and `splits_comp_of_splits` in Data/Polynomial/Splits and use the last one to golf `splits_of_algHom`, `splits_of_isScalarTower` (introduced in # 8609).

+ add three new lemmas `mem_range_x_of_minpoly_splits` to simplify the construction of `IntermediateField.algHomEquivAlgHomOfIsAlgClosed` and `Algebra.IsAlgebraic.algHomEquivAlgHomOfIsAlgClosed`, remove the `IsAlgClosed` condition and rename. ~~They could be moved to an earlier file but I refrain from doing that. (#find_home says it's already in the right place)~~

+ golf `primitive_element_iff_algHom_eq_of_eval` from # 8609, using a new lemma `IsIntegral.minpoly_splits_tower_top` for the last step.

+ make `integralClosure_algEquiv_restrict` (from # 8714) computable and rename to `AlgEquiv.mapIntegralClosure` to follow camelCase naming convention and enable dot notation.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alreadydone** [comment]
cc @acmepjz @riccardobrasca

**Comment by riccardobrasca** [comment]
Can you please fix the build error?

File: Mathlib/FieldTheory/IsAlgClosed/Basic.lean
Comment (xroblot) [inline Mathlib/FieldTheory/IsAlgClosed/Basic.lean:411]: You need to move the `[IsAlgClosed]` on line 401 to the statement of `Algebra.IsAlgebraic.range_eval_eq_rootSet_minpoly` otherwise it is an (unnecessary) hypothesis in the next results. (I don't know how to produce a diff to show that)

File: Mathlib/RingTheory/Adjoin/Field.lean
Comment (xroblot) [inline Mathlib/RingTheory/Adjoin/Field.lean:89]: That's a great addition.

**Review by xroblot — APPROVED** [review]

File: Mathlib/FieldTheory/Adjoin.lean
Comment (riccardobrasca) [inline Mathlib/FieldTheory/Adjoin.lean:856]: Since you are at it, maybe protect also this one?

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Review by alreadydone — COMMENTED** [review]
I addressed the reviews but also added more lemmas, so I'll wait for another approval or otherwise I'll merge tomorrow. Thanks!

**Comment by riccardobrasca** [comment]
It still looks good to me, thanks!

**Comment by riccardobrasca** [comment]
bors merge

File: Mathlib/FieldTheory/IsAlgClosed/Basic.lean
Comment (xroblot) [reply Mathlib/FieldTheory/IsAlgClosed/Basic.lean:411]: yeah, I was quite surprised you missed that. I should have known better. At least it's clearer now (:
Sorry for the noise

**Pull request closed without merging**
